src/Google/Http/REST.php

```diff
@@ -93,10 +94,21 @@ public static function decodeHttpResponse(
   ) {
     $body = (string) $response->getBody();
     $code = $response->getStatusCode();
+    $result = null;
 
     // return raw response when "alt" is "media"
     $isJson = !($request && 'media' == $request->getQuery()->get('alt'));
-    $result = $isJson ? $response->json() : $body;
+
+    // set the result to the body if it's not set to anything else
+    if ($isJson) {
+      try {
+        $result = $response->json();
+      } catch (ParseException $e) {
```

Comment:
Also, whatever the non-JSON is gets returned, so nothing super valuable is thrown away (IMHO), just whatever parse error was discovered by `json_decode`.